### PR TITLE
[AC-2499] Add permission checks on bulk actions menu

### PR DESCRIPTION
**Pull request by eliykat**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

The bulk actions menus (in the individual and organization vaults) did not consistently check that the user had permissions to carry out the selected action. I suspect we have some bugs around this today, but it mostly worked because owners and admins could manage all collections and items.

However, Flexible Collections v1 changes restrict owner/admin permissions, so you're more likely to come across the situation where you don't have permissions to carry out the bulk action.

I've reviewed all the bulk actions and added permission checks where required - usually one of the `collectionView` getters (`canDelete`, `canEdit`) or `cipher.edit`. These incorporate feature flag checks where required, so the changes in this PR are not themselves feature flagged.

Notably, a user must have sufficient permissions for **all** items/collections selected. We are not allowing any partial success.

Other, minor changes:
* remove the header on error toasts, per design feedback. This includes existing error toasts in these components for consistency
* change the bulk access error from `You have not selected anything` to `You have not selected any collections` - minor bugfix for the situation where the user has selected items but no collections.

## Code changes

- **file.ext:** Description of what was changed and why

## Screenshots
Screenshot 2024-04-25 at 10 32 04 AM

Screenshot 2024-04-25 at 10 31 48 AM

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

**Review by shane-melton — DISMISSED** [review]
This looks good to me!

**Review by shane-melton — APPROVED** [review]

**Comment by eliykat** [comment]
Converting to draft while I clarify requirements and potentially add additional flagging.

**Comment by eliykat** [comment]
@shane-melton I've made further changes after our discussions and going round in circles a bit.

The problem is that the individual vault should only evaluate individual permissions, not admin permissions. I considered just not touching the individual vault at all, because really the v1 collection management setting only limits admin actions in Admin Console. However, we're already using `collectionView.canEdit`/`canDelete` in the individual vault, and that includes admin permissions, so there's already leakage between the two contexts. Ignoring individual vault here would probably just cause unintended consequences/bugs/questions from QA.

Instead I went with your suggestion of updating `CollectionView` to only use individual permissions, and `CollectionAdminView` to use admin permissions. While I was at it, I removed the pre-FC logic in these getters. I think this should basically fix all the leakage here and make sure this ticket (and subsequent changes) work as intended.

One downside here is that many methods in the org `VaultComponent` use `CollectionView` in their signature, but receive a `CollectionAdminView`, so it's not clear which getters are being used. That can't be updated without breaking other things (e.g. `VaultItemEvent`) which are used in both contexts. I guess that's how polymorphism works though.

It will require QA before merge, but I'd come to the conclusion that that would be required anyway.

File: apps/web/src/app/vault/org-vault/vault.component.ts [hunk 1]
```diff
@@ -971,11 +971,16 @@ export class VaultComponent implements OnInit, OnDestroy {
   }
 
   async restore(c: CipherView): Promise<boolean> {
-    if (!(await this.repromptCipher([c]))) {
+    if (!c.isDeleted) {
       return;
     }
 
-    if (!c.isDeleted) {
+    if (!c.edit && !this.organization.allowAdminAccessToAllCollectionItems) {
```
Comment (eliykat) [inline apps/web/src/app/vault/org-vault/vault.component.ts:978]: I realised that we need to take into account collection management settings for operations on ciphers.

File: apps/web/src/locales/en/messages.json
Comment (shane-melton) [inline apps/web/src/locales/en/messages.json:1]: question: Not sure why this shows up in your diff; seems unrelated?

**Review by shane-melton — APPROVED** [review]
I think this worked out nicely with updating the `CollectionView` to only consider individual permissions and will help clarify the delineation. 

Also, thanks for removing those `org.flexibleCollections` checks!

File: apps/web/src/locales/en/messages.json
Comment (eliykat) [reply apps/web/src/locales/en/messages.json:1]: Agree, I'll remove it.

**Comment by eliykat** [comment]
I'm going to wait until #9100 is merged, as that will probably be merged first and will cause conflicts.

**Comment by eliykat** [comment]
@shane-melton Hopefully the final set of changes:
1. merge in `main` after various other PRs have been merged, particularly #9100
2. feature flag the checkbox changes
3. feature flag all the permission checks (maybe overkill, but it's safest to ensure new code isn't run)
4. feature flag the refactor to `collectionView.canDelete` and `collectionView.canEdit`. Now FC MVP logic is the default, but if the FC v1 flag is on it'll use the new code which splits individual vs. admin permissions.

File: apps/web/src/app/vault/individual-vault/vault.component.ts [hunk 2]
```diff
@@ -847,13 +859,27 @@ export class VaultComponent implements OnInit, OnDestroy {
...
+      collections == null ||
+      collections.every((c) =>
+        c.canDelete(
+          organizations.find((o) => o.id == c.organizationId),
+          flexibleCollectionsV1Enabled,
+        ),
       );
+    const canDeleteCiphers = ciphers == null || ciphers.every((c) => c.edit);
+
+    if (flexibleCollectionsV1Enabled && (!canDeleteCollections || !canDeleteCiphers)) {
```
Comment (eliykat) [inline apps/web/src/app/vault/individual-vault/vault.component.ts:878]: This sucks, but it's the clearest way I can find to express it. Other suggestions are welcome. Keeping it in 1 line made prettier start to do weird things and put brackets in unexpected places.

**Review by shane-melton — APPROVED** [review]
New changes look good to go, thank you!

**Review by shane-melton — APPROVED** [review]
